src/Components/Components/src/Auth/AuthorizeRouteView.cs

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.RenderTree;
+
+namespace Microsoft.AspNetCore.Components
+{
+    /// <summary>
+    /// Combines the behaviors of <see cref="AuthorizeView"/> and <see cref="RouteView"/>,
+    /// so that it displays the page matching the specified route but only if the user
+    /// is authorized to see it.
+    ///
+    /// Additionally, this component supplies a cascading parameter of type <see cref="Task{AuthenticationState}"/>,
+    /// which makes the user's current authentication state available to descendants.
+    /// </summary>
+    public sealed class AuthorizeRouteView : RouteView
+    {
+        /// <summary>
+        /// The content that will be displayed if the user is not authorized.
+        /// </summary>
+        [Parameter] public RenderFragment<AuthenticationState> NotAuthorized { get; set; }
+
+        /// <summary>
+        /// The content that will be displayed while asynchronous authorization is in progress.
+        /// </summary>
+        [Parameter] public RenderFragment Authorizing { get; set; }
+
+        /// <inheritdoc />
+        protected override void Render(RenderTreeBuilder builder)
+        {
+            // TODO: Consider merging the behavior of CascadingAuthenticationState into this
+            // component (i.e., rendering the CascadingValue directly) to avoid the extra
+            // layer of component nesting and eliminate CascadingAuthenticationState as public API.
```

Comment:
+1 for removing as a public API.

Reply:
You probably saw, but the comment was obsoleted and removed in a later commit.

This PR retains `CascadingAuthenticationState` as public API because it's what you'd use if you wanted to use authorization *without* URL-based routing. A likely scenario for that would be a native app.

However `CascadingAuthenticationState` no longer shows up in the project templates, so that's an improvement.